pkg/skaffold/build/kaniko/sources/sources.go

```diff
@@ -81,7 +81,39 @@ func podTemplate(cfg *latest.KanikoBuild, args []string) *v1.Pod {
 						SecretName: cfg.PullSecretName,
 					},
 				},
-			}},
+			},
+			},
 		},
 	}
+
+	if cfg.MountDockerConfig {
+		mountPath := constants.DefaultKanikoDockerConfigPath
+		if cfg.DockerConfigPath != "" {
+			mountPath = cfg.DockerConfigPath
```

Comment:
I believe the mountPath should still be `constants.DefaultKanikoDockerConfigPath` here. If a user passes in `cfg.DockerConfigPath`, that should be the path to config.json on their local machine. If you pass in this path to `v1.VolumeMount` later on, it will mount in whatever's at that path on the cluster node, not on the machine.

To get around this, it might make sense to first create a secret from `cfg.DockerConfigPath`, and then mount that into the pod.  For reference, this is similar to how kaniko handles the pull [secret](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/build/kaniko/secret.go#L32) as well.

Let me know if that didn't make sense! :)

Reply:
Ok, that makes sense, I've taken the opposite approach here by assuming the `config.json` was already available as a secret in the cluster.  I'll get the code updated to handle this in the same way as the pull secret.